public/app/features/datasources/state/actions.ts

```diff
@@ -111,9 +111,18 @@ export function loadDataSources(): ThunkResult<void> {
   };
 }
 
-export function loadDataSource(id: number): ThunkResult<void> {
+export function loadDataSource(uid: string): ThunkResult<void> {
   return async (dispatch) => {
-    const dataSource = (await getBackendSrv().get(`/api/datasources/${id}`)) as DataSourceSettings;
+    // Check if this is an old id format!
+    const id = parseInt(uid, 10);
+    if (id) {
+      const ds = (await getBackendSrv().get(`/api/datasources/${id}`)) as DataSourceSettings;
+      if (ds?.uid) {
+        console.log('TODO... redirect!', ds.uid);
```

Comment:
must be something cleaner than this!  any ideas?

Reply:
I  can try figure this out, pushing something soon

Reply:
just saw this -- pushed somehting but it breaks with:
![image](https://user-images.githubusercontent.com/705951/117527338-15b83300-af80-11eb-8251-1b617673b996.png)

Reply:
it's working now, was a bit tricky to fix